src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/WitnessesDraftMidEventHandler.java

```diff
@@ -99,22 +86,29 @@ public PreSubmitCallbackResponse<AsylumCase> handle(PreSubmitCallbackStage callb
         return response;
     }
 
-    private String buildWitnessFullName(WitnessDetails witnessDetails) {
-        return (witnessDetails.getWitnessName() + " " + witnessDetails.getWitnessFamilyName()).trim();
-    }
-
-    private void clearWitnessFieldsPreemptively(AsylumCase asylumCase) {
-        WITNESS_N_FIELD.forEach(field -> asylumCase.write(field, new WitnessDetails("", "")));
-        WITNESS_LIST_ELEMENT_N_FIELD.forEach(field -> asylumCase.write(field, new DynamicMultiSelectList()));
-        WITNESS_N_INTERPRETER_CATEGORY_FIELD.forEach(field -> asylumCase.write(field, Collections.emptyList()));
-    }
-
-    private void clearWitnessInterpreterLanguageFields(AsylumCase asylumCase) {
-        DynamicList dummyDynamicList = new DynamicList("");
-        WITNESS_N_INTERPRETER_SPOKEN_LANGUAGE.forEach(field ->
-                asylumCase.write(field, new InterpreterLanguageRefData(dummyDynamicList, Collections.emptyList(), "")));
-        WITNESS_N_INTERPRETER_SIGN_LANGUAGE.forEach(field ->
-            asylumCase.write(field, new InterpreterLanguageRefData(dummyDynamicList, Collections.emptyList(), "")));
+    /**
+     * Breaks witnessDetails collection down into individual witness fields (witness1, witness2 etc.) and generates the
+     * individual dynamicMultiSelectList fields for each witness (witnessListElement1, witnessListElement2 etc.)
+     * @param asylumCase The asylum case
+     * @param witnesses  The value of the witnessDetails field (collection)
+     * @return           The asylum case enriched with the generated fields
+     */
+    protected AsylumCase decentralizeWitnessCollection(AsylumCase asylumCase,
+                                                       List<IdValue<WitnessDetails>> witnesses) {
+        int i = 0;
+        while (i < witnesses.size()) {
+
+            String fullName = buildWitnessFullName(witnesses.get(i).getValue());
+
+            asylumCase.write(WITNESS_N_FIELD.get(i), witnesses.get(i).getValue());
+            asylumCase.write(WITNESS_LIST_ELEMENT_N_FIELD.get(i),
+                new DynamicMultiSelectList(
+                    Collections.emptyList(),
+                    List.of(new Value(fullName, fullName))
+                ));
+            i++;
+        }
+        return asylumCase;
```

Comment:
You have a similar piece of code in InterpreterLanguageUtils.java. Why not reuse it?

Reply:
The one in InterpreterLanguagesUtils was getting used in a past PRs but I think it had no usages in this latest version of the code.
I cleaned that up and deleted that method + another method in InterpreterLanguagesUtils that were unused.